src/widget.h

```diff
@@ -242,6 +246,7 @@ class TextInputWidget : public LeafWidget {
 	size_t cursor_pos;                ///< Position of the cursor in the text.
 	int value_width;                  ///< Width of the image or the string.
 	int value_height;                 ///< Height of the image or the string.
+	bool has_focus;                   ///< The widget currently owns the keyboard focus.
```

Comment:
You're writing explanations that don't explain anything, basically `bool X; // (do/get) X;`

The explanation should explain X without using X, as people reading the comments wouldn't know what X means (or they wouldn't read the comments).